api/v1/db/json/notes-dao.js

```diff
@@ -72,24 +76,75 @@ const getNotes = query => new Promise((resolve, reject) => {
 });
 
 /**
- * @summary Return a specific pet by unique ID
+ * @summary Return a specific note by noteID
  * @function
- * @param {string} id Unique pet ID
- * @returns {Promise} Promise object represents a specific pet
+ * @param {string} noteID id of the note in the form: '{studentID}-{number}'
+ * @returns {Promise} Promise object represents a specific note
  */
-// const getPetById = id => new Promise((resolve, reject) => {
-//   try {
-//     const rawPets = appRoot.require('/tests/unit/mock-data.json').pets;
-//     const rawPet = _.find(rawPets, { ID: id });
-//     if (!rawPet) {
-//       resolve(undefined);
-//     } else {
-//       const serializedPet = SerializedPet(rawPet);
-//       resolve(serializedPet);
-//     }
-//   } catch (err) {
-//     reject(err);
-//   }
-// });
-
-module.exports = { getNotes };
+const getNoteByID = noteID => new Promise((resolve, reject) => {
+  try {
+    const studentID = noteID.split('-')[0];
+    const studentDirPath = `${dbDirectoryPath}/${studentID}`;
+
+    const rawNote = readJSONFile(`${studentDirPath}/${noteID}.json`);
+    if (!rawNote) {
+      resolve(null);
+    }
+    rawNote.source = localSourceName;
+
+    const serializedNote = serializeNote(rawNote);
+    resolve(serializedNote);
+  } catch (err) {
+    reject(err);
+  }
+});
+
+/**
+ * @summary Create a new note
+ * @function
+ * @param body
+ * @returns {Promise} Promise object representing the new note
+ */
+const postNotes = body => new Promise((resolve, reject) => {
+  try {
+    const {
+      note, studentID, creatorID,
+    } = body;
```

Comment:
As we discussed, we should probably validate request body and return a `400` if the body is invalid.

Reply:
Currently, express-openapi should handle errors for invalid request bodies, such as having incorrect types for fields or not having required fields. From our discussion, you suggested checking for any unknown properties in the request body and responding with an error if any were encountered. Apparently there's a field that can be used in the swagger document for something like this. From https://swagger.io/specification/#properties:
> * additionalProperties - Value can be boolean or object. Inline or referenced schema MUST be of a Schema Object and not a standard JSON Schema. Consistent with JSON Schema, additionalProperties defaults to true.

So if, for some object, `additionalProperties` is `false`, an openapi error will be thrown if any additional properties are encountered. Should we use this? I couldn't find anything related to this in the jsonapi specification so I think it would be up to us. What are your thoughts?

Reply:
For example, if the object is something like
```json
"data": {
  "attributes": {
    "extraField": "extraValue"
  }
}
```
`path` will be `data.attributes`, `location` will be `body`, and `message` will be `should NOT have additional properties`. Our default handler would respond with something like this for `detail`: `"Error in path: 'data.attributes', location: 'body', message: 'should NOT have additional properties'"` which I think is clear enough, although it wouldn't be hard to customize the message if we need to. @tsoliangwu0130 what do you think?

Reply:
From our discussion, this default handler won't respond with the name of the additional property so we should add a specific rule for this.